src/system-health/health_checker/sysmonitor.py

```diff
@@ -0,0 +1,634 @@
+        
+    def task_worker(self):
+        while not self.task_stopping_event.is_set():
+            try:
+                self.task_worker_statedb()
+            except Exception as e:
+                logger.log_error("task_worker_statdb exited- {}, retry".format(str(e)))
+            time.sleep(1)
+    
+
+    def task_run(self):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_process = multiprocessing.Process(target=self.task_worker)
+        self.task_process.start()
+
+    def task_notify(self, msg):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_queue.put(msg)
+        time.sleep(0.001)
+
+    def task_stop(self):
+        self.task_stopping_event.set()
+        self.task_process.join()
+
+
+class MonitorSystemBusTask(object):
+    def __init__(self):
+        self.task_queue = mpmgr.Queue()
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def __init__(self,myQ):
+        self.task_queue = myQ
+        self.task_process = None
+        self.task_stopping_event = multiprocessing.Event()
+
+    def OnJobRemoved(self, id, job, unit, result):
+        #print('{}: Job Removed: {}, {}, {} '.format( id, job, unit, result))
+        if result == "done":
+            #logger.log_info("Received systemd service event for {}".format(unit))
+            timestamp="{}".format(datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"))
+            msg={"unit": unit,"evt_src":"sysbus","time":timestamp}
+            self.task_notify(msg)
+            return
+
+    #Sub process for listening the systemd event on dbus
+    def subscribe_sysbus(self):
+        import dbus
+        from gi.repository import GObject
+        from dbus.mainloop.glib import DBusGMainLoop
+
+        DBusGMainLoop(set_as_default=True)
+
+        bus = dbus.SystemBus()
+        systemd = bus.get_object('org.freedesktop.systemd1', '/org/freedesktop/systemd1')
+        manager = dbus.Interface(systemd, 'org.freedesktop.systemd1.Manager')
+
+        manager.Subscribe()
+        manager.connect_to_signal('JobRemoved', self.OnJobRemoved)
+
+        loop = GObject.MainLoop()
+        loop.run()
+
+    def task_worker(self):
+        #logger.log_info("Start Listening to systemd bus (pid {0})".format(os.getpid()))
+        update_proc_name('sysmonitor_sysbus')
+        
+        while not self.task_stopping_event.is_set():
+            self.subscribe_sysbus()
+            time.sleep(1)
+
+    def task_run(self):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_process = multiprocessing.Process(target=self.task_worker)
+        self.task_process.start()
+
+    def task_notify(self, msg):
+        if self.task_stopping_event.is_set():
+            return
+        self.task_queue.put(msg)
+        time.sleep(0.001)
+
+    def task_stop(self):
+        self.task_stopping_event.set()
```

Comment:
maybe you need stop the event loop?

Reply:
when the event loop runs, as ProcessTaskBase in src/sonic-py-common/sonic_py_common/task_base.py has task_stop which actually kills the subprocess if not joined(exited) on time.

    def task_stop(self):
        # Signal the process to stop
        self.task_stopping_event.set()

        # Wait for the process to exit
        self._task_process.join(self._stop_timeout_secs)

        # If the process didn't exit, attempt to kill it
        if self._task_process.is_alive():
            os.kill(self._task_process.pid, signal.SIGKILL)

        if self._task_process.is_alive():
            return False